transformations/greetings_and_farewells/transformation.py

```diff
@@ -0,0 +1,116 @@
+import itertools
+import random
+import re
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+# Setting up a tuple of possible replaceable greetings and farewells
+GREETINGS = (
+    "Hi",
+    "Hello",
+    "Hey",
+    "Howdy",
+    "Greetings",
+    "Good morning",
+    "Good afternoon",
+    "Good evening",
+    "What's up",
+    "Sup",
+)
+
+# Creating a tuple of sentences that can't be simply replaced by a short greeting
+SPECIAL_GREETINGS = (
+    "It's nice to meet you.",
+    "It's great to meet you.",
+    "Pleased to meet you.",
+    "How are you?",
+    "How are you doing?",
+    "How is it going on?",
+    "How is it going?",
+)
+
+FAREWELLS = (
+    "Goodbye",
+    "Bye bye",
+    "Bye",
+    "See you soon",
+    "See you",
+    "See ya",
+    "Best regards",
+    "Have a nice day",
+    "Have a great day",
+    "Have a nice weekend",
+    "Good night",
+    "I gotta go",
+)
+
+# Compiling regex
+GREETINGS_REGEX = (
+    re.compile("good (morning|afternoon|evening)", re.IGNORECASE),
+    re.compile(r"\b(what's up|sup)\b(\?|)", re.IGNORECASE),
+    re.compile(r"\b(hi|hello|hey|howdy)\b", re.IGNORECASE),
+)
+
+SPECIAL_GREETINGS_REGEX = (
+    re.compile(
+        "(it('s| is| was) |)(a |)(nice|great|pleased|pleasure) to meet (you|u)(.|)",
+        re.IGNORECASE,
+    ),
+    re.compile("how( are|'re|) (you|u) doin(g|'|)\?", re.IGNORECASE),
+    re.compile("how( is|'s|) it going( on|)\?", re.IGNORECASE),
+    re.compile("how( have|'ve|) you been\?", re.IGNORECASE),
+)
+
+FAREWELLS_REGEX = (
+    re.compile("(good night|goodbye|goodnight)", re.IGNORECASE),
+    re.compile("see (you|u|ya)( soon| later| tomorrow|)", re.IGNORECASE),
+    re.compile("have a (great|nice|good) (day|night|week|weekend)", re.IGNORECASE),
+    re.compile("best regards", re.IGNORECASE),
+    re.compile(
+        r"\b(by+e+)+\b", re.IGNORECASE
+    ),  # it matches 'bye', 'bye bye', 'byyyyyyye', 'byeeeee', etc.
+)
+
+
+def greetings_and_farewells(text, seed=0, max_output=1):
+    random.seed(seed)
```

Comment:
I believe `random.seed()` needs to be called each time before `random.choice()`. I would recommend generating a random integer as a seed for each `random.choice()` call using this seed to increase diversity in the test examples while remaining reproducible.